packages/compiler/core/parser.ts

```diff
@@ -1547,50 +1550,64 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
     const id = parseIdentifier();
 
     parseExpected(Token.OpenBrace);
-    let from, to;
-    let proj1, proj2;
-    if (token() === Token.Identifier) {
-      proj1 = parseProjection();
 
-      if (token() === Token.Identifier) {
-        proj2 = parseProjection();
+    const projectionMap = new Map<string, ProjectionNode>();
+    const projections: ProjectionNode[] = [];
+    while (token() === Token.Identifier) {
+      const projection = parseProjection();
+      if (projection.direction !== "<error>") {
```

Comment:
Does `direction` still make sense as the name?

Reply:
It's a bit off. I wanted to avoid renaming property and breaking and also couldn't come up with something better.

Reply:
don't think thats a big deal so whatever but would also having a `pre: boolean` property instead help?

Reply:
Thought about this but it ends up being helpful to have a single differentiating "key" so leaving it.